lib/saluki-config/src/lib.rs

```diff
@@ -442,6 +442,65 @@ impl ConfigurationLoader {
             })
         }
     }
+
+    /// Configures a [`GenericConfiguration`] that is suitable for tests.
+    ///
+    /// This configures the loader with the following defaults:
+    ///
+    /// - configuration from a JSON file
+    /// - configuration from environment variables
+    ///
+    /// If `enable_dynamic_configuration` is true, a dynamic configuration sender is returned.
+    ///
+    /// This is generally only useful for testing purposes, and is exposed publicly in order to be used in cross-crate testing scenarios.
+    pub async fn for_tests(
+        file_values: Option<serde_json::Value>, env_vars: Option<&[(String, String)]>,
+        enable_dynamic_configuration: bool,
+    ) -> (GenericConfiguration, Option<tokio::sync::mpsc::Sender<ConfigUpdate>>) {
+        let json_file = tempfile::NamedTempFile::new().expect("should not fail to create temp file.");
+        let path = &json_file.path();
+        let json_to_write = file_values.unwrap_or(serde_json::json!({}));
+        serde_json::to_writer(&json_file, &json_to_write).expect("should not fail to write to temp file.");
+
+        let mut loader = ConfigurationLoader::default().try_from_json(path);
+        let mut maybe_sender = None;
+        if enable_dynamic_configuration {
+            let (sender, receiver) = tokio::sync::mpsc::channel(1);
+            loader = loader.with_dynamic_configuration(receiver);
+            maybe_sender = Some(sender);
+        }
+
+        static ENV_MUTEX: OnceLock<std::sync::Mutex<()>> = OnceLock::new();
+
+        let guard = ENV_MUTEX.get_or_init(|| std::sync::Mutex::new(())).lock().unwrap();
+
+        if let Some(pairs) = env_vars.as_ref() {
+            for (k, v) in pairs.iter() {
+                std::env::set_var(k, v);
```

Comment:
I think we'd want to prefix all of these with `TEST` so that the caller doesn't have to.